### PR TITLE
ResetOffset - Preliminary README Docs

### PR DESCRIPTION
**Pull request by travis-minke-sap**
Subtask of #588 to provide some initial user documentation README information.

## Proposed Changes

- 🎁  Added user level information next to CRD definition.
- 🎁  Added developer level information next to Command source code.
- 🧽  Cleaned up current "distributed" KafkaChannel README which was a bit dated.

**Notes**
- The ResetOffset Controller is still "inactive" and has not been added to any Channels/Sources/etc.
- This PR is not intended to comprise the FULL set of documentation for the ResetOffset feature - still need to determine whether other external docs are required, etc.  This is just a beginning at the repo level README information.

### CODE REVIEW

File: pkg/common/commands/resetoffset/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Data-Plane implementation, the intent is that the common
+[ConsumerManager](../../consumer/consumer_manager.go) will be used. This
+implementation already provides the expected ConsumerGroup lifecycle management
+and locking control.
+
+
+
+
+
+
```
Comment (eric-sap) [inline pkg/common/commands/resetoffset/README.md:118]: What's all this empty space for?

**Review by eric-sap — COMMENTED** [review]
Looks good!  Some minor comments.

File: config/command/resources/RESET_OFFSET.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
+# ResetOffset Command CRD
+
+The ResetOffset "command" provides the ability to reposition the Kafka
+ConsumerGroup Offsets of a particular Knative resources (Subscription, Trigger,
```
[the remark this answers is not in the document]
Comment (travis-minke-sap) [reply config/command/resources/RESET_OFFSET.md:4]: good catch - will actually change `resources` to `resource` i think.

File: config/command/resources/RESET_OFFSET.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+experiencing lengthy exponential back-offs, allowing the locks to expire and
+another instance to begin.
+
+While you can create the ResetOffset instance in any namespace, it makes sense
+to use the same namespace as the Subscription "ref" of which you wish to
+reposition the Offsets. If no `spec.ref.namespace` value is provided it will
+default to the ResetOffset namespace.
+
+The `spec.offset.time` is a string that can be one of **"earliest"**,
+**"latest"** or a valid RFC3339 format timestamp. As you might expect, the
```
[the remark this answers is not in the document]
Comment (travis-minke-sap) [reply config/command/resources/RESET_OFFSET.md:65]: haha - just channeling my inner Shakespear (poorly) - will remove ; )

File: pkg/channel/distributed/README.md [hunk 4]
```diff
@@ -10,58 +10,34 @@ This repo falls under the
...
-repository) already contains a Kafka implementation, so why invest the time in
-building another one? At the time this project was begun, and still today, the
-reference Kafka implementation does not provide the scaling characteristics
-required by a large and varied use case with many different Topics and
-Consumers. That implementation is based on a single choke point that could
-easily allow one Topic's traffic to impact the throughput of another Topic. It
-is also not horizontally scalable as it only supports a single instance of the
-dispatcher/consumer. Further, no ordering guarantees on the consumption side are
-provided which is required in certain use cases.
+The Knative "consolidated" KafkaChannel implementation already provides an
```
[the remark this answers is not in the document]
Comment (travis-minke-sap) [reply pkg/channel/distributed/README.md:19]: 👍

File: pkg/channel/distributed/README.md [hunk 5]
```diff
@@ -10,58 +10,34 @@ This repo falls under the
...
-default "kafka" implementation. The project which previously existed in the open
-source [kyma-incubator](https://github.com/kyma-incubator/), but was moved into
-it's current home here in the
-[knative-sandbox](https://github.com/knative-sandbox) The focus is now on
-bringing the implementation into alignment with the other knative channel
-implementations.
+## Status
+
+Significant work has recently gone into aligning the two implementations from a
+CRD, configuration, authorization, and code- sharing perspective, in order to
```
[the remark this answers is not in the document]
Comment (travis-minke-sap) [reply pkg/channel/distributed/README.md:35]: doh!   line break followed by reformat shenanigans - thanks!

File: pkg/common/commands/resetoffset/README.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+3 - Stop all related ConsumerGroups in the Dispatcher Replicas.
+
+4 - Reposition the Offsets of all ConsumerGroup Partitions.
+
+5 - Re-Start all related ConsumerGroups in the Dispatcher Replicas.
+```
+
+The reconciler will continue to process a ResetOffset until is has `Succeeded`.
+It is careful to only ever reposition the ConsumerGroup Offsets a single time to
+prevent the back/forth repositioning in certain failure scenarios.
```
[the remark this answers is not in the document]
Comment (travis-minke-sap) [reply pkg/common/commands/resetoffset/README.md:75]: 👍

File: pkg/common/commands/resetoffset/README.md (hunk 1 above)
Comment (travis-minke-sap) [reply pkg/common/commands/resetoffset/README.md:118]: I get paid by the number of lines of code i write. 🙃

**Comment by eric-sap** [comment]
/lgtm
thanks!

**Comment by tajpouria** [comment]
Where are the RestOffset docs now?

**Pull request merged**